### ISSUE TITLE
[Regression]Maximum number of Paging requests are not getting triggered from AMF

### ISSUE THREAD

**Issue by NeelimaKalpana** [magma/magma#11064]
Your Environment

- Version :
- Affected Component: Access Gateway
- Affected SubcomponentAGW service
- Deployment Environment: Vagrant (AGW)

### Describe the Issue
Only two paging request is triggered by AGW towards a gNB, 3rd and 4th last attempt of paging is missing to send the Service Request = MT from UE . So as Part of TC9b , we are receiving Service Request = Signaling instead of MT and actual TC expectation is not getting triggered.

**To Reproduce**
1. Please execute TC9b : Verify_Paging_reponse_in_last_paging_Retry
2. TC is failing as AMF is not sending the expected number of paging attempts .

**Expected behavior**
AMF should try for maximum number of paging retries to hit the scenario.

[https://app.zenhub.com/files/170803235/87beeafa-8742-4ec7-b0d2-149378fe1ec7/download](https://app.zenhub.com/files/170803235/87beeafa-8742-4ec7-b0d2-149378fe1ec7/download)

**Comment by moinuddin1980** [comment]
As per TS 24.501 section 5.6.2 Paging Procedure, there should  not be any re-transmissions for paging messages from AMF towards UE to be sent.
Hence, expecting paging re-transmission of  5 times is not as per expectation of from specs.
![image](https://user-images.githubusercontent.com/93516965/148582289-5c358f6d-a921-4b3d-8874-3bf39fddd25d.png)

Also, it is clearly evident from the pcap timing and syslogs that the the AMF and SMF are trying to release the PDU session as per instruction from UE. i-e. Only after Service Request and service accept procedure between UE and AMF, the UE send PDU session release request. Hence, the AMF is sending gRPC to SMF for clearing the PDU session.
Thus, the behavior of AMF and SMF is correct.
![image](https://user-images.githubusercontent.com/93516965/148583440-afb41f1e-ea56-4f58-a693-81011059515c.png)

![image](https://user-images.githubusercontent.com/93516965/148584148-ec59ec77-5e33-4519-8c77-879fae7b7774.png)

Hence, the TC expectation need to modified, as paging message re-transmission is not as per specs.

**Comment by moinuddin1980** [comment]
More info can be found regarding ServiceType IE in Service Request message  in TS 24.501 section "5.6.1.2 Service request procedure initiation"
Request you to check whether proper ServiceType IE is set always in Service Request for any paging related TC.

**Comment by NeelimaKalpana** [comment]
Hi Arun ,
Please share your opinion w.r.t TC9b.
Or Please modify the verdict accordingly.

**Comment by arun-magma** [comment]
The no of paging tries to 3+ attempts can give enough attempts for UE to latch on to the network with SR (MT),  to be in-fact no of retries should be a configurable value and need to let the network operators to decide based on network density where UE is located.
Generally Diff levels of paging strategy need to be followed as below.
- TA
- TAI
- Global paging to all TACs in the Network

**Comment by NeelimaKalpana** [comment]
Hi Arun,

Please share the working pcaps of TC9b- Maximum paging retries testing.

**Comment by arun-magma** [comment]
@NeelimaKalpana : Now T3513=4s and N3513=1 could trigger on 2nd paging with the Service_Request=Mobile termination, But Magma is not responding with Service Accept, Plz check.

[Pcaps]
[https://app.zenhub.com/files/170803235/ca09a858-6f7f-4665-b863-9a1f80ba5274/download](https://app.zenhub.com/files/170803235/ca09a858-6f7f-4665-b863-9a1f80ba5274/download)

[Service_accept missing]
![Screen Shot 2022-01-31 at 8.36.33 PM.png](https://images.zenhubusercontent.com/5f934b19f9d04269a3f065cd/b3d35c67-b513-48cc-8334-cb07ea861ef8)

**Comment by rupa455wavelabs** [comment]
Hi @arun-magma, 
Developers are looking into the service accept issue, PR is in pending state. we will test it once the PR got merged into master branch.

Thanks

**Comment by arun-magma** [comment]
@NeelimaKalpana : Where as TC9a is to check only the paging attempts Magma retries, no SR_MT is planned in it so with our SR_MT TC should pass.

As TC9b with SR_MT covers the TC9a at some point we may remove TC9a.

**Comment by NeelimaKalpana** [comment]
Hi Arun,

Thanks for the update.

**Comment by NeelimaKalpana** [comment]
Validated and closing the issue with required logs and pcaps.
[https://app.zenhub.com/files/170803235/79eecbf2-e92b-446f-86fa-1e8e29ae5d73/download](https://app.zenhub.com/files/170803235/79eecbf2-e92b-446f-86fa-1e8e29ae5d73/download)
